data/sea/54-zebra.h

```diff
@@ -154,68 +185,142 @@ static ierror_msg_t zebra_translate
     return 0;
 }
 
-static int64_t zebra_translate_table
+/* Translate Zebra enum tags to Icicle bool flags, e.g. is_some, is_right, etc.
+   We only allow at most two variants since Icicle's runtime representation
+   of enums is unsuitable for more.
+ */
+static uint64_t zebra_translate_column_tags
+    ( const int64_t variant_count
+    , void **dst
+    , const int64_t *src )
+{
+    assert (variant_count <= 2);
+
+    *dst = src;
+    return 1;
+}
+
+
+static uint64_t zebra_translate_column_nested
     ( anemone_mempool_t *mempool
-    , iint_t elem_start
-    , iint_t count
+    , int64_t elem_start
+    , int64_t elem_count
     , void **dst
-    , const zebra_table_t *table )
+    , const int64_t *indices
+    , const zebra_table_t *src )
 {
-    int64_t         offset = 0;
-    int64_t         cols_count  = table->column_count;
-    zebra_column_t *cols = table->columns;
-
-    //printf ("zebra_translate_table: elem_start = %d, count = %d\n", elem_start, count);
-    for (int64_t i = 0; i != cols_count; ++i) {
-        //printf ("zebra_translate_table: offset = %d\n", offset);
-        offset += zebra_translate_column (mempool, elem_start, count, dst + offset, cols + i);
+    /* Only read nested tables of type binary now.
+       FIXME: slice out nested tables, when we need to support them. */
+    assert (src->tag == ZEBRA_TABLE_BINARY);
+
+    int64_t offset = 1;
+
+    for (int i = 0; i != elem_count; ++i) {
+        const int64_t elem_index = elem_start + i;
+        const int64_t table_start = indices[elem_index] - indices[0];
+        const int64_t table_row_count = indices[elem_index + 1] - indices[elem_index];
+
+        /* read the inner tables */
+        void **target = anemone_mempool_alloc (mempool, elem_count * 8);
+        zebra_translate_table (mempool, table_start, table_row_count, target + i, src);
+        *dst = target;
+    }
+    return offset;
+}
+
+static uint64_t zebra_translate_named_column
+    ( anemone_mempool_t *mempool
+    , int64_t elem_start
+    , int64_t elem_count
+    , void **dst
+    , const zebra_named_columns_t *src )
+{
+    int64_t offset = 0;
+    const int64_t columns_count = src->count;
+    const zebra_column_t *columns = src->columns;
+
+    for (int64_t i = 0; i != columns_count; ++i) {
+        offset += zebra_translate_column (mempool, elem_start, elem_count, dst + offset, columns + i);
     }
 
     return offset;
 }
 
-static int64_t zebra_translate_column
+static uint64_t zebra_translate_column
```

Comment:
I think you should probably stick with signed numbers, unsigned arithmetic is not a fun place to be